Assets/MixedRealityToolkit/Providers/Hands/BaseHandVisualizer.cs

```diff
@@ -176,6 +176,16 @@ public void OnHandMeshUpdated(InputEventData<HandMeshInfo> eventData)
             {
                 Mesh mesh = handMeshFilter.mesh;
 
+                // On some platforms, mesh length counts may change as the hand mesh is updated.
+                // In order to update the vertices when the array sizes change, the mesh
+                // must be cleared per instructions here:
+                // https://docs.unity3d.com/ScriptReference/Mesh.html
+                if ((mesh.vertices?.Length ?? 0) != 0 &&
```

Comment:
[mesh.vertices](https://docs.unity3d.com/ScriptReference/Mesh-vertices.html) returns a **copy** of a mesh's vertices (crazy, right?). So, you may want to cache them once before the conditional, rather than potentially creating two copies. Also, I'm pretty Unity guarantees mesh verticies, indicies, etc. are never null (they can be empty) so you could remove those checks too.

Reply:
Thanks for calling that out - I ended up caching the last vertex array (to avoid having an additional copy each time).